Modules/Forum/classes/class.ilForumNotification.php

```diff
@@ -438,15 +438,19 @@ public static function mergeThreadNotificiations($merge_source_thread_id, $merge
         );
 
         $res_2 = $ilDB->queryF(
-            'SELECT user_id FROM frm_notification 
+            'SELECT DISTINCT user_id FROM frm_notification 
 		WHERE frm_id = %s 
 		AND  thread_id = %s
 		 ORDER BY user_id ASC',
             array('integer', 'integer'),
             array(0, $merge_target_thread_id)
         );
 
-        $users_already_notified = $ilDB->fetchAssoc($res_2);
+        $users_already_notified = array();
+        while ($users_row = $ilDB->fetchAssoc($res_2)) {
+            $users_already_notified[$users_row['user_id']] = $users_row['user_id'];
+        }
+
         while ($row = $ilDB->fetchAssoc($res)) {
             if (in_array($row['user_id'], $users_already_notified)) {
```

Comment:
@nmatuschek Can you please use `isset` here? Using a map and `isset` is much faster than `in_array` applied on large lists.